src/yunohost/log.py

```diff
@@ -0,0 +1,436 @@
+        Write or rewrite the metadata file with all metadata known
+        """
+
+        filename = os.path.join(self.path, self.name + METADATA_FILE_EXT)
+        with open(filename, 'w') as outfile:
+            yaml.safe_dump(self.metadata, outfile, default_flow_style=False)
+
+    @property
+    def name(self):
+        """
+        Name of the operation
+        This name is used as filename, so don't use space
+        """
+        if self._name is not None:
+            return self._name
+
+        name = [self.started_at.strftime("%Y%m%d-%H%M%S")]
+        name += [self.operation]
+        if self.related_to:
+            if isinstance(self.related_to[0], tuple):
+                name += [self.related_to[0][1]]
+            else:
+                name += self.related_to[0].values()
+        self._name = '-'.join(name)
+        return self._name
+
+    @property
+    def metadata(self):
+        """
+        Dictionnary of all metadata collected
+        """
+
+        data = {
+            'started_at': self.started_at,
+            'operation': self.operation,
+        }
+        if self.related_to is not None:
+            data['related_to'] = self.related_to
+        if self.ended_at is not None:
+            data['ended_at'] = self.ended_at
+            data['success'] = self._success
+            if self.error is not None:
+                data['error'] = self._error
+                # TODO: detect if 'extra' erase some key of 'data'
+        data.update(self.extra)
+        return data
+
+    def success(self):
+        """
+        Declare the success end of the unit operation
+        """
+        self.close()
+
+    def error(self, error):
+        """
+        Declare the failure of the unit operation
+        """
+        return self.close(error)
+
+    def close(self, error=None):
+        """
+        Close properly the unit operation
+        """
+        if self.ended_at is not None or self.started_at is None:
+            return
+        if error is not None and not isinstance(error, basestring):
+            error = str(error)
+        self.ended_at = datetime.now()
+        self._error = error
+        self._success = error is None
+        if self.logger is not None:
+            self.logger.removeHandler(self.file_handler)
+
+        is_api = msettings.get('interface') == 'api'
+        desc = _get_description_from_name(self.name)
+        if error is None:
+            if is_api:
+                msg = m18n.n('log_link_to_log', name=self.name, desc=desc)
+            else:
+                msg = m18n.n('log_help_to_get_log', name=self.name, desc=desc)
+            logger.info(msg)
+        else:
+            if is_api:
+                msg = "<strong>" + m18n.n('log_link_to_failed_log',
+                                        name=self.name, desc=desc) + "</strong>"
+            else:
+                msg = m18n.n('log_help_to_get_failed_log', name=self.name, desc=desc)
+            logger.warning(msg)
```

Comment:
About this I don't know either what we should do ... I just tested a dummy cert-install like this : 

```
 > yunohost domain cert-install gruh.schpof
Info: Now attempting install of certificate for domain gruh.schpof!
Error: No DNS 'A' record found for gruh.schpof. You need to make your domain name point to your machine to be able to install a Let's Encrypt certificate! (If you know what you are doing, use --no-checks to disable those checks.)
Warning: The operation 'Install Let's encrypt certificate on 'gruh.schpof' domain' has failed ! To get help, please share the full log of this operation using the command 'yunohost log display 20180725-154526-letsencrypt_cert_install-gruh.schpof --share'
```
So if I'm a use not paying much attention, what I understand is that I should upload the log and get some help on the forum/chat about this. But the true meaningful message here is that there is no A record in the DNS and the user should check its DNS config. (That's especially relevant for the webadmin where, if the user don't click on the top bar, he'll only see the very last message)

Thing is that we have at least two kind of errors. 'Expected' errors like this (e.g. invalid arguments, invalid DNS conf, ...) where we shouldn't print this message, and 'Unexpected' errors (e.g. unexpected exception or failed app install) where we want to show this. But not sure how to easily differentiate the two :/